lib/rpmsg/rpmsg_virtio.c

```diff
@@ -223,10 +223,14 @@ static int rpmsg_virtio_wait_remote_ready(struct rpmsg_virtio_device *rvdev)
 	while (1) {
 		status = rpmsg_virtio_get_status(rvdev);
 		/* Busy wait until the remote is ready */
+#ifndef VIRTIO_SLAVE_ONLY
 		if (status & VIRTIO_CONFIG_STATUS_NEEDS_RESET) {
 			rpmsg_virtio_set_status(rvdev, 0);
 			/* TODO notify remote processor */
-		} else if (status & VIRTIO_CONFIG_STATUS_DRIVER_OK) {
+			continue;
+		}
+#endif /*!VIRTIO_MASTER_ONLY*/
+		if (status & VIRTIO_CONFIG_STATUS_DRIVER_OK) {
```

Comment:
Should this driver ok checking only apply to virtio slave?

Reply:
For me both, the slave and the master can call it, but now regarding the implementation the function is only called by slave in rpmsg_init_vdev . 
So perhaps a best solution is  to simply suppress this piece of code.

Reply:
On the slave side, if the master hasn't set the driver_ok, the slave side cannot finish the init as it will need to use the vrings. Agree both sides can check the status, but as the status is set by master for now, and driver_ok bit can only set by master, only slave need to check the bit.